deploy/docker-compose/README.md

```diff
@@ -0,0 +1,98 @@
+# Docker Compose Deployment
+
+## Quick Start
+
+This is a simple example that shows you how to connect `nm-vllm` with production monitoring via Prometheus and Grafana via Docker Compose.
+
+## Install
+
+Make sure you have Docker and Docker Compose installed.
+- [`docker`](https://docs.docker.com/engine/install/)
+- [`docker compose`](https://docs.docker.com/compose/install/linux/#install-using-the-repository)
+
+> Note: `nm-vllm` requires a GPU. Make sure you have installed the [Nvidia Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html).
+
+## Deploy
+
+### Launch
+
+The following spins up an `nm-vllm` instance, a Prometheus server, and a Grafana server:
+
+```bash
+docker compose up
+```
+
+### Simulate Client Requests
+
+Install requirements for client:
+```bash
+pip install -r client/requirements.txt
+```
+
+Download some sample data:
+```bash
+wget https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/resolve/main/ShareGPT_V3_unfiltered_cleaned_split.json
+```
+
+Launch clients:
+
+```bash
+python3 client/client.py --dataset-path ShareGPT_V3_unfiltered_cleaned_split.json --request-rate 1.0
+```
+
+### Monitor with Grafana
+
+There are three steps to monitor your `nm-vllm` service:
+
+1. Login to Grafana Client
+    - Navigate to the Grafana Client [`http://localhost:3000`](http://localhost:3000) and log in with the default username (`admin`) and password (`admin`).
+
+2. Add Prometheus Data Source
+    - Navigate to [`http://localhost:3000/connections/datasources/new`](http://localhost:3000/connections/datasources/new) and select Prometheus.
+    - On Prometheus configuration page, add `http://prometheus:9090` as the `Prometheus Server URL` in `Connection`.
+    - Click `Save & Test`. You should get a green check saying "Successfully queried the Prometheus API."
+
+3. Navigate to [`http://localhost:3000/dashboard/import`](http://localhost:3000/dashboard/import), upload `grafana.json`, selecting the `prometheus` datasource. You will see a dashboard that looks like the following:
+
+</br>
+<div text-align="center">
+    <img width="75%" src="assets/grafana-dashboard.png" />
+</div>
```

Comment:
I think you could markdown this as
```suggestion
![Grafana Dashboard](assets/grafana-dashboard.png)
```

but perhaps you are looking for control over image sizing

Reply:
If you are looking for control over image sizing, we might consider downscaling the image so we don't transfer bytes in fidelity that we don't intend to show anyway. Also a `jpg` would probably yield a smaller image size if you care about such things. `png` requires extra bytes for handling transparency that `jpg` does not and I don't think your dashboard image would include transparency.